data/meson.build

```diff
@@ -24,9 +24,16 @@ install_data(['metadata.xml'],
   install_dir : join_paths(datadir, 'fwupd', 'remotes.d', 'fwupd')
 )
 
-install_data(['90-fwupd-devices.rules'],
-  install_dir : join_paths(udevdir, 'rules.d')
-)
+if get_option('daemon')
+  udevdir = get_option('udevdir')
+  if udevdir == ''
+    udev = dependency('udev')
```

Comment:
I'd prefer to keep all dependency selection logic at the top level meson.build so it's easy to find what options are introducing new dependencies.  Would you mind to move conditional udev dependency there?